mesonbuild/coredata.py

```diff
@@ -218,6 +224,8 @@ def validate_value(self, value: T.Any) -> T.Union[str, OctalInt]:
         return OctalInt(self._validate_value(value))
 
     def toint(self, valuestring: T.Union[str, OctalInt]) -> int:
+        if isinstance(valuestring, OctalInt):
+            valuestring = str(valuestring)
```

Comment:
This typing is simply a lie, we only ever invoke it if `isinstance(value, str)`. If it is already an int, we do not need to convert it to a str and then back to an int by way of base 8. :D We just use it as is.

Actually I think the only way to get an OctalInt is from getting the sanity-checked value via an instantiated umask option, which isn't ever going to be fed back into itself anyway...

Whatever. Point is, I am pretty sure this should just be `def toint(self, valuestring: str) -> int:`